.github/workflows/solidity-foundry.yml

```diff
@@ -0,0 +1,56 @@
+name: Solidity Foundry
+on: [pull_request]
+
+env:
+  FOUNDRY_PROFILE: ci
+
+jobs:
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.changes.outputs.src }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721 # v2.10.2
+        id: changes
+        with:
+          filters: |
+            src:
+              - 'contracts/**/*'
+              - '.github/workflows/solidity-foundry.yml'
+
+  tests:
```

Comment:
Given that we're only introducing the OCR2DROracle contract here, I think we can add the publicABI check scripts in.

Reply:
I think we might need to tune those a bit, probably best to add them in a future PR